BrainPortal/app/models/cluster_task.rb

```diff
@@ -1566,10 +1566,33 @@ def submit_cluster_job
       # Queue the job on the cluster and return true, at this point
       # it's not our 'job' to figure out if it worked or not.
       self.addlog("Cluster command: #{job.qsub_command}") if self.user.has_role? :admin_user
-      jobid              = scir_session.run(job)
-      self.cluster_jobid = jobid
-      self.status_transition(self.status, "Queued")
-      self.addlog("Queued as job ID '#{jobid}'.")
+      begin
+        jobid              = scir_session.run(job)
+        self.cluster_jobid = jobid
+        self.status_transition(self.status, "Queued")
+        self.addlog("Queued as job ID '#{jobid}'.")
+      rescue => ex
+        # When the task is executed in a VM, it may not be submitted
+        # right away when no VMs are available. In such a case, method
+        # scir_cloud.run will raise an exception. We need to mask this
+        # exception and put the task back to status "New" so that
+        # submission will be attempted again by the Bourreau later
+        # on. VMs may become available for the task when other tasks
+        # complete or new VMs are started.
+        raise ex unless ex.message.include?("Cannot match task to VM.") 
```

Comment:
maybe check on the class of the exception instead of the message in the exception? 

```ruby
  if ex.is_a?(ScirCloudSubmitException)
  addlog()
end
```